buildattr64/buildattr64.rst

```diff
@@ -0,0 +1,904 @@
+
+.. table:: Attributes Section
+
+    +---------------------+----------------------------+-------+-----------+------+------------+
+    | Name                | Type                       | Flags | Alignment | Info | Entry Size |
+    +=====================+============================+=======+===========+======+============+
+    | ``.ARM.attributes`` | ``SHT_AARCH64_ATTRIBUTES`` | 0x0   | 1         | 0    | 0          |
+    +---------------------+----------------------------+-------+-----------+------+------------+
+
+A consumer may not assume the natural alignment of a data type such as ``uint32``.
+
+The syntactic structure of an attributes section is::
+
+   <format-version: <version-letter>>
+   [ <uint32: subsection-length> NTBS: vendor-name
+     <bytes: vendor-data>
+   ]*
+
+*Format-version* describes the format of the following data. It is a
+single byte. Only version 'A' (0x41) is defined. This field exists to
+allow future changes in format.
+
+*Subsection-length* is a 4-byte integer in the byte order of the ELF
+file. It encodes the length of the subsection, including the length
+field itself, the vendor name string and its terminating NUL byte, and
+the following vendor data. It gives the offset from the start of this
+subsection to the start of the next one.
+
+*Vendor-name* is a NUL-terminated byte string (NTBS) like a C-language
+literal string.
+
+No requirements are placed on the format of private vendor data. The
+format of the public attributes subsections (which will have a
+pseudo-vendor name with a prefix of “aeabi”) are described below.
+
+Attributes that record data about the compatibility of this
+relocatable object file with other relocatable object files must be
+recorded in a public "aeabi" prefixed subsection.
+
+Attributes meaningful only to the producer must be recorded in a
+private vendor subsection. These must not affect compatibility between
+relocatable object files. A producer that does not recognize the
+private vendor may skip over the private vendor subsection.
+
+Formally, there are no constraints on the order or number of vendor
+subsections. A consumer can collect the public ("aeabi" prefixed)
+attributes in a single pass over the section, then all of its private
+data in a second pass.
+
+Public aeabi prefixed subsections
+=================================
+
+About public tags
+-----------------
+
+Subsections containing a vendor name with a prefix of "aeabi" are
+public attributes defined by this specification. All public aeabi
+sections have the same format, which includes a header. The header
+will indicate whether the section is required or optional.
+
+A consumer must be able to parse the header of all subsections that
+have a prefix of "aeabi". If the header is optional then a consumer
+may skip the subsection if they do not recognize the full vendor-name.
+
+Default values for public tags
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The effect of omitting a public tag in file scope is identical to
```

Comment:
It is unclear to me whether this is true only for optional subsections, only for mandatory subsections,  or for both, optional and mandatory and subsections.

Reply:
I'll see if I can make the mandatory part clearer. I think an examples section in the rationale would help too. All tags get an implicit default of 0 whether in a mandatory or optional section.

A subsection should be mandatory if a static linker must be able to understand at least one of the tags defined in it to produce correct output. By making the subsection mandatory a static linker is expected to give an error message if it can't process the subsection.

If the subsection is optional and a static linker doesn't understand either the subsection or a tag within it then it can just skip over it. The output will be correct, if not optimal.

Reply:
Continuing the discussion regarding default values per tag:
If we have the role that omitting a tag is like having it with value "0", then this is all we need in order to handle old object files(?) 
Then it is not clear what is the part of having a default value per tag has to do with handling old object files?